### DISCUSSION TITLE
How do you create a ring shape using iModel.js

### DISCUSSION THREAD

**Discussion by PeterJBassett** [iTwin/itwinjs-core#4601] [Q&A] [answered]
Hello, I am working on the FutureOn FieldTwin to iTwin connector, and have a requirement to draw various types of shapes into an iModel.

My question is - is there a way to create a ring shape ie. a flat circle with a circular hole in the middle. (like a flattened torus).
I have looked into the 3d geometry web-pages, but can't work out how to do it using the library.

**Comment by bbastings** [comment, marked as answer]
You'll want to create a sub-type of CurveCollection called a ParityRegion. The ParityRegion will have 2 child loops, each created from a 360 degree Arc3d. When creating the smaller/hole Loop, set the isInner property to true.

See [CurveCollection](https://www.itwinjs.org/learning/geometry/curvecollection/) for more information.

**Reply by PeterJBassett** [reply]
Hello @bbastings - thank so much for your help that has worked. 
This is the code I ended up with:
```
        const centre = Point3d.create(0, 0, 0);
        const outer = Loop.create(Arc3d.createXYEllipse(centre, radiusOuter, radiusOuter));
        const inner = Loop.create(Arc3d.createXYEllipse(centre, radiusInner, radiusInner))
        inner.isInner = true;
        iModelShape = ParityRegion.createLoops([ outer, inner ]);
```
